# 파이썬 2일차

## 중간실습 - input, 포매팅 사용
1. 포매팅을 사용하여 이름을 입력받아 화면에 'Hello, 이름'을 출력하세요
2. 섭씨 온도(C)를 입력받아 화씨 온도(F)로 변환하여 출력하세요(변환공식: F = C * 9/5 + 32)
3. 금액을 입력받아, 10%, 20%, 30% 팁을 계산하여 출력하세요
4. 문자열을 입력받아 그 문자열의 길이를 구하세요
5. 원의 반지름을 입력받아  원의 면적(πr²)과 둘레(2πr)를 출력하세요 (π는 3.14159로 가정)
6. 정수 하나를 입력받아 그 수의 2배, 3배, 4배의 값을 한줄에 출력하세요
7. 이름을 입력받아 이름의 문자들 사이에 공백을 두고 출력하세요
8. 소숫점을 포함한 숫자를 입력받아 소숫점 이하를 제거하고 정수 부분만 출력하세요


In [ ]:
# 1. 포매팅을 사용하여 이름을 입력받아 화면에 'Hello, 이름'을 출력하세요
name = input("이름을 입력하세요: ")
print(f"Hello, {name}!")

Hello, 김영욱!


In [ ]:
# 2. 섭씨 온도(C)를 입력받아 화씨 온도(F)로 변환하여 출력하세요(변환공식: F = C * 9/5 + 32)
C = float(input("섭씨 온도를 입력하세요: "))
F = C * 9/5 + 32
print(f"화씨 온도는: {F}도 입니다.")

화씨 온도는: 89.6도 입니다.


In [4]:
# 3. 금액을 입력받아, 10%, 20%, 30% 팁을 계산하여 출력하세요
amount = int(input("금액을 입력하세요: "))
print(f"10% 팁: {amount * 0.1}, 20% 팁: {amount * 0.2}, 30% 팁: {amount * 0.3}")

10% 팁: 1000.0, 20% 팁: 2000.0, 30% 팁: 3000.0


In [5]:
# 4. 문자열을 입력받아 그 문자열의 길이를 구하세요
string = input("문자열을 입력하세요: ")
print(f"'{string}'의 길이는: {len(string)}입니다.")

'경기도 용인시 기흥구'의 길이는: 11입니다.


In [2]:
# 5. 원의 반지름을 입력받아  원의 면적(πr²)과 둘레(2πr)를 출력하세요 (π는 3.14159로 가정)
r = float(input("원의 반지름을 입력하세요: "))
pi = 3.14159
면적 = pi * r ** 2
둘레 = 2 * pi * r
print(f"반지름: {r}, 면적: {면적}, 둘레: {둘레:.4f}")

반지름: 4.5, 면적: 63.617197499999996, 둘레: 28.2743


In [ ]:
# 6. 정수 하나를 입력받아 그 수의 2배, 3배, 4배의 값을 한줄에 출력하세요
num = int(input("Enter an integer: "))
print(f"2x: {num*2}, 3x: {num*3}, 4x: {num*4}")

2x: 8, 3x: 12, 4x: 16


In [ ]:
# 7. 이름을 입력받아 이름의 문자들 사이에 공백을 두고 출력하세요
name = input("Enter your name: ")
spaced_name = " ".join(name)
print(spaced_name)

김 영 욱


In [ ]:
# 8. 소숫점을 포함한 숫자를 입력받아 소숫점 이하를 제거하고 정수 부분만 출력하세요
float_num = float(input("소수점이 포함된 숫자를 입력하세요: "))
int_part = int(float_num)
print(f"{float_num}의 정수 부분은 {int_part}입니다")

3.14의 정수 부분은 3입니다


## 5. 리스트(list) 자료형

리스트(list)는 파이썬에서 여러 값들을 순서대로 저장하는 데 사용되는 자료형입니다.  
리스트는 대괄호 [] 안에 요소들을 쉼표로 구분하여 나열함으로써 생성됩니다.  
리스트는 다양한 타입의 데이터를 하나의 리스트에 저장할 수 있으며, 같은 데이터 타입이 아니어도 됩니다.

리스트의 주요 특징:

변경 가능(Mutable): 리스트는 생성된 후에도 그 내용을 변경할 수 있습니다. 요소를 추가, 삭제, 또는 변경하는 것이 가능합니다.  
순서가 있음(Ordered): 리스트에 추가된 요소는 순서를 가지며, 이 순서는 변경되지 않습니다. 따라서 인덱스를 통해 접근할 수 있습니다.  
중복 허용: 리스트 내에서 같은 값을 가진 요소를 여러 개 가질 수 있습니다.

리스트명 = [요소1, 요소2, 요소3, 요소4, .....] # 리스트 요소에는 데이터 타입에 의한 제한은 없습니다.

list_a

| Life | is | too | short, | You | need | Python |
|------|----|-----|--------|-----|------|--------|
| 0    | 1  | 2   | 3      | 4   | 5    | 6      |
| -7   | -6 | -5  | -4     | -3  | -2   | -1     |


In [5]:
a = 'Life is too short, You need Python'
b = 'pithon'

list_a = a.split()
list_b = [['Life', 'is'], ['too', 'short']]
list_c = [1, 2, 3, 4, 5]
list_d = [6, 7, 8, 9, 10]
list_e = list(b) # ['p', 'i', 't', 'h', 'o', 'n']

In [7]:
list_b = list(b)
list_b[1] = 'y'

print(list_b)


['p', 'y', 't', 'h', 'o', 'n']


### 1. 리스트 인덱싱, 슬라이싱

In [ ]:
list_a[0]

'Life'

In [ ]:
list_a[:4]

['Life', 'is', 'too', 'short,']

In [ ]:
list_a[4:]

['You', 'need', 'Python']

In [ ]:
list_c[0] + list_c[1] # 1 + 2

3

In [ ]:
list_b[0]

['Life', 'is']

In [ ]:
list_b[0][0]

'Life'

### 2. 리스트 연산

In [ ]:
list_c + list_d

[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

In [ ]:
list_c * 3 # 리스트는 중복값을 가질 수 있음

[1, 2, 3, 4, 5, 1, 2, 3, 4, 5, 1, 2, 3, 4, 5]

In [301]:
sum(list_c*3) # 리스트 합계

45

### 3. 리스트 길이 구하기

In [ ]:
print(len(list_c))

5


### 4. 리스트 값 수정, 삭제

In [ ]:
list_c = [1, 2, 3, 4, 5]

list_c[0] = 10

print(list_c)

[10, 2, 3, 4, 5]


In [8]:
a = 'pithon'
a = list(a)

a[1] = 'y'

print(a)

['p', 'y', 't', 'h', 'o', 'n']


In [ ]:
del list_c[0]

print(list_c)

[2, 3, 4, 5]


### 5, 리스트 함수

#### 1. append

In [ ]:
list_c = [1, 2, 3, 4, 5]

list_c.append(6)

print(list_c) # 맨 뒤에 추가

[1, 2, 3, 4, 5, 6]


#### 2. sort

In [ ]:
list_c = [1, 5, 3, 2, 4]

list_c.sort()

print(list_c) # 정렬

[1, 2, 3, 4, 5]


In [ ]:
list_d = ["a", "c", "d", "b"]

list_d.sort()

print(list_d)

['a', 'b', 'c', 'd']


#### 3. reverse

In [ ]:
list_c = [1, 5, 3, 2, 4]

list_c.reverse()

print(list_c) # 정렬 후 역순으로 정렬하는 것이 아님

[4, 2, 3, 5, 1]


#### 4. index

In [ ]:
list_c = [1, 5, 3, 2, 4]

print(list_c.index(3)) # 위치값 반환, 문자열 index 처럼 요소값이 없으면 오류

2


#### 5. insert

In [ ]:
list_c = [1, 2, 3, 4, 6]

list_c.insert(4, 5) # 4번째 위치에 5를 삽입

print(list_c)

[1, 2, 3, 4, 5, 6]


#### 6. remove

In [ ]:
list_c = [1, 2, 1, 2]

list_c.remove(1) # 리스트에서 첫번째로 나오는 1을 제거

print(list_c)

[2, 1, 2]


#### 7. pop

In [ ]:
list_c = [1, 2, 3, 4, 5]

print(list_c.pop()) # 리스트의 맨 마지막 요소를 리턴하고 삭제

print(list_c.pop(1)) # 리스트의 2번째 요소를 리턴하고 삭제

print(list_c)

5
2
[1, 3, 4]


#### 8. count

In [ ]:
list_c = [1, 2, 1, 2]

print(list_c.count(1))

2


#### 9. extend

In [ ]:
list_c = [1, 2, 3]

list_c.extend([4, 5]) # 반복 가능한 개체(리스트, 튜플, 문자열, 집합, 딕셔너리)

print(list_c)

[1, 2, 3, 4, 5]


#### 10. min

In [302]:
list_c = [1, 2, 3, 4, 5]

min(list_c)

1

#### 11. max

In [303]:
max(list_c)

5

#### 12. 정리

| 함수      | 설명                                                         |
|---------|------------------------------------------------------------|
| append  | 리스트 끝에 항목을 추가합니다.  list_a.append(6)                                         |
| sort    | 리스트의 요소를 오름차순 또는 내림차순으로 정렬합니다. 설정을 통해 정렬 순서를 변경할 수 있습니다. list_a.sort(), list_a.sort(reverse=True)    |
| reverse | 리스트의 요소 순서를 뒤집습니다.  list_a.reverse()                                     |
| index   | 지정한 요소가 처음으로 나타나는 위치의 인덱스를 반환합니다. 요소가 리스트에 없으면 오류가 발생합니다.  list_a.index(3)|
| insert  | 지정한 위치에 요소를 삽입합니다. 위치 인덱스는 리스트의 시작부터 0으로 계산됩니다.  list_a.insert(4, 5) # 4번째 위치에 5를 삽입             |
| remove  | 리스트에서 첫 번째로 나오는 지정한 값을 삭제합니다. 요소가 리스트에 없으면 오류가 발생합니다.  list_a.remove(1) # 리스트에서 첫번째로 나오는 1을 제거      |
| pop     | 지정한 위치에 있는 요소를 리스트에서 제거하고, 그 값을 반환합니다. 위치를 지정하지 않으면, 마지막 요소를 제거하고 반환합니다.  list_a.pop()  |
| count   | 리스트에서 지정한 값이 나타나는 횟수를 반환합니다.  list_a.count(1)                             |
| extend  | 리스트 끝에 다른 리스트(또는 반복 가능한 다른 요소)의 모든 항목을 추가합니다.   list_a.extend([4, 5])            |
| min     | 리스트 내의 최소값을 반환합니다. 숫자 또는 알파벳 순서로 최소값을 찾을 수 있습니다.   min(list_a)           |
| max     | 리스트 내의 최대값을 반환합니다. 숫자 또는 알파벳 순서로 최대값을 찾을 수 있습니다.   max(list_a)           |


#### 13. 리스트 실습

문제 1
1. 1부터 10까지의 숫자를 포함하는 리스트를 생성하세요
2. 생성한 리스트에서 세 번째 요소를 출력하세요
3. 리스트의 마지막 요소를 출력하세요

In [30]:
# 문제 1
# 1. 1부터 10까지의 숫자를 포함하는 리스트를 생성하세요.
numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

# 2. 생성한 리스트에서 세 번째 요소를 출력하세요.
print(numbers[2])

# 3. 리스트의 마지막 요소를 출력하세요.
print(numbers[-1])
print(numbers[9])


3
10
10


In [36]:
num = list(range(1, 11))

print(num[2])
print(num[-1])

3
10


문제 2
1. 다음 문자열을 요소로 하는 리스트를 생성하세요 "apple", "banana", "cherry"
2. "banana" 대신 "blueberry"를 넣어서 리스트를 수정하세요
3. 리스트 끝에 "orange"를 추가하세요
4. 리스트의 두 번째 위치에 "kiwi"를 십압하세요

In [31]:
# 문제 2
# 1. 다음 문자열을 요소로 하는 리스트를 생성하세요: "apple", "banana", "cherry"
fruits = ["apple", "banana", "cherry"]

# 2. "banana" 대신 "blueberry"를 넣어서 리스트를 수정하세요.
fruits[1] = "blueberry"

# 3. 리스트 끝에 "orange"를 추가하세요.
fruits.append("orange")

# 4. 리스트의 두 번째 위치에 "kiwi"를 삽입하세요.
fruits.insert(1, "kiwi")

print(fruits)

['apple', 'kiwi', 'blueberry', 'cherry', 'orange']


In [38]:
a = ["apple", "banana", "cherry"]

a[1] = 'blueberry'
a.append('orange')
a.insert(1, 'kiwi')

print(a)

['apple', 'kiwi', 'blueberry', 'cherry', 'orange']


문제 3
1. 다음 리스트를 사용하세요: [3, 1, 4, 1, 5, 9, 2]
2. 리스트를 오름차순으로 정렬하세요.
3. 리스트에서 최대값과 최소값을 찾아 출력하세요.
4. 리스트의 길이를 출력하세요.
5. 리스트에서 "1"의 개수를 세어 출력하세요.

In [32]:
# 문제 3
# 다음 리스트를 사용하세요: [3, 1, 4, 1, 5, 9, 2]
numbers = [3, 1, 4, 1, 5, 9, 2]

# 2. 리스트를 오름차순으로 정렬하세요.
numbers.sort()
print("오름차순 정렬:", numbers)

# 3. 리스트에서 최대값과 최소값을 찾아 출력하세요.
print("최대값:", max(numbers))
print("최소값:", min(numbers))

# 4. 리스트의 길이를 출력하세요.
print("리스트 길이:", len(numbers))

# 5. 리스트에서 "1"의 개수를 세어 출력하세요.
print('"1"의 개수:', numbers.count(1))

오름차순 정렬: [1, 1, 2, 3, 4, 5, 9]
최대값: 9
최소값: 1
리스트 길이: 7
"1"의 개수: 2


## 6. 튜플(tuple) 자료형


튜플(tuple)은 파이썬에서 여러 값을 순서대로 저장하는 데 사용되는 불변(immutable) 자료형입니다.  
튜플은 소괄호 () 안에 요소들을 쉼표로 구분하여 나열함으로써 생성되며, 괄호를 생략할 때도 있습니다.  
튜플의 주요 특성은 한 번 생성되면 그 내용을 변경할 수 없다는 것입니다.  
이 불변성 때문에 튜플은 리스트보다 상대적으로 공간 효율적이고, 해싱 가능하여 딕셔너리의 키로 사용될 수 있습니다.

튜플의 주요 특징:

불변(Immutable): 튜플은 생성된 후에는 그 내용을 변경할 수 없습니다. 요소를 추가하거나 삭제하거나 변경하는 것이 불가능합니다. (sort, remove, pop과 같은 내장 함수가 없다)  
순서가 있음(Ordered): 튜플에 추가된 요소는 순서를 가지며, 이 순서는 변경되지 않습니다. 따라서 인덱스를 통해 접근할 수 있습니다.  
중복 허용: 튜플 내에서 같은 값을 가진 요소를 여러 개 가질 수 있습니다.


튜플명 = (요소1, 요소2, 요소3, 요소4, .....)

In [ ]:
t1 = ()
t2 = (1, 2, 3, 4)
t3 = (1, 2, 3, ("a", "b", "c"))

In [ ]:
# del t2[0] # 튜플은 요소값을 지울 수 없다

TypeError: 'tuple' object doesn't support item deletion

In [ ]:
# t2[0] = 10 # 튜플은 요소값을 변경할 수 없다

TypeError: 'tuple' object does not support item assignment

### 1. 튜플 인덱싱, 슬라이싱

In [ ]:
t2[0]

1

In [ ]:
t3[-1]

('a', 'b', 'c')

In [ ]:
t3[-1][0]

'a'

In [ ]:
t2[1:]

(2, 3, 4)

### 2. 튜플 연산

In [ ]:
t2 + t3

(1, 2, 3, 4, 1, 2, 3, ('a', 'b', 'c'))

In [ ]:
t2 * 2

(1, 2, 3, 4, 1, 2, 3, 4)

### 3. 튜플 길이 구하기

In [ ]:
print(len(t2))

4


## 7. 딕셔너리(dictionary) 자료형


딕셔너리(dictionary)는 키(key)와 값(value) 쌍을 요소로 가지는 파이썬의 내장 자료형입니다.  
각 키는 딕셔너리 내에서 유일해야 하며, 변하지 않는(immutable) 데이터 타입이어야 합니다.  
따라서 키로는 문자열, 숫자, 튜플 등이 사용될 수 있지만, 리스트는 사용될 수 없습니다.  
딕셔너리는 중괄호 {}를 사용하여 생성하며, 키:(콜론)값 형태로 요소들을 나열합니다.

딕셔너리의 주요 특징:

변경 가능(Mutable): 딕셔너리는 생성된 후에도 그 내용을 변경할 수 있습니다. 요소를 추가, 삭제, 또는 변경하는 것이 가능합니다. (키 값은 변경 불가, 삭제 후 추가)  
순서 없음(Unordered): 파이썬 3.7 이전 버전에서는 딕셔너리 요소의 순서가 보장되지 않았습니다. 하지만, 파이썬 3.7부터는 삽입된 순서를 유지합니다. 그러나 인덱스를 통한 접근은 불가능하며, 키를 사용하여 접근합니다.  
키의 유일성: 각 키는 딕셔너리 내에서 유일해야 하며, 한 딕셔너리 내에 같은 키를 두 번 사용할 수 없습니다.

딕셔너리명 = {key1:velue1, key2:velue2, ....}

| 키        | 값     |
|-----------|-------|
| name      | Alice |
| age       | 25    |
| language  | Python|

In [268]:
my_empty_dict = {}
my_dict = {'name': 'Alice', 'age': 25, 'language': 'Python'}

### 1. 딕셔너리 사용

In [ ]:
print(my_dict['name']) # key를 사용해 value 값 얻기

Alice


### 2. 딕셔너리 요소 추가

In [ ]:
my_dict['country'] = 'Wonderland'

In [ ]:
print(my_dict)

{'name': 'Alice', 'age': 25, 'language': 'Python', 'country': 'Wonderland'}


### 3. 딕셔너리 value 변경

In [238]:
my_dict['age'] = 26

In [239]:
print(my_dict)

{'name': 'Alice', 'age': 26, 'language': 'Python', 'country': 'Wonderland'}


### 4. 딕셔너리 요소 삭제

In [240]:
del my_dict['language']

In [241]:
print(my_dict)

{'name': 'Alice', 'age': 26, 'country': 'Wonderland'}


### 5. 딕셔너리 함수

In [242]:
my_dict = {'name': 'Alice', 'age': 25, 'language': 'Python', 'country':'Wonderland'}

#### 1. keys

In [249]:
a = my_dict.keys()

print(a) # key 값을 모아 dict_keys를 반환

dict_keys(['name', 'age', 'language', 'country'])


#### 2. values

In [252]:
b = my_dict.values()

print(b) # value 값을 모아 dict_values를 반환

dict_values(['Alice', 25, 'Python', 'Wonderland'])


#### 3. items

In [251]:
c = my_dict.items()

print(c) # key, value 값을 모아 dict_items를 반환

dict_items([('name', 'Alice'), ('age', 25), ('language', 'Python'), ('country', 'Wonderland')])


dict_keys, dict_values, dict_items은 파이썬 3.0 이후 버전에서 추가된 것 따로 리스트로 변환 하지 않더라고 반복문 사용 가능

#### 4. clear

In [253]:
my_dict.clear()

print(my_dict) # 딕셔너리 안의 모든 요소를 삭제

{}


#### 5. get

In [257]:
my_dict = {'name': 'Alice', 'age': 25, 'language': 'Python', 'country':'Wonderland'}

print(my_dict.get('name'))
print(my_dict['name'])

print(my_dict.get('phoneNum')) # 없는 값을 가져올때 None
# print(my_dict['phoneNum']) # 없는 값을 가져올때 에러

Alice
Alice
None


#### 6. in

In [258]:
'name' in my_dict # 해당 키가 딕셔너리 안에 있는지 확인

True

In [259]:
'phoneNum' in my_dict

False

#### 7. 정리

| 함수 | 설명 |
|------|------|
| `dict.get(key[, default])` | 주어진 키에 대한 값을 반환합니다. 키가 딕셔너리에 없으면 `default`를 반환합니다. `default`가 제공되지 않으면 `None`을 반환합니다. |
| `dict.keys()` | 딕셔너리의 키들을 반환하는 뷰(view)를 반환합니다. |
| `dict.values()` | 딕셔너리의 값들을 반환하는 뷰(view)를 반환합니다. |
| `dict.items()` | 딕셔너리의 키-값 쌍을 튜플로 가지는 뷰(view)를 반환합니다. |
| `dict.update([other])` | 딕셔너리를 다른 딕셔너리 객체나 키-값 쌍의 반복 가능한(iterable) 객체로부터 업데이트합니다. |
| `dict.pop(key[, default])` | 주어진 키에 해당하는 항목을 딕셔너리에서 제거하고, 그 값을 반환합니다. 키가 딕셔너리에 없으면 `default`가 반환됩니다. `default`가 제공되지 않고 키가 없는 경우 KeyError가 발생합니다. |
| `dict.popitem()` | 딕셔너리에서 마지막으로 삽입된 키-값 쌍을 제거하고 반환합니다. 딕셔너리가 비어있으면 KeyError가 발생합니다. |

## 8. 집합(set) 자료형

집합(set) 자료형은 파이썬에서 중복을 허용하지 않고, 순서가 없는 요소들의 모임을 다룰 때 사용합니다.  
집합은 중괄호 {}를 사용하여 생성하며, 집합 내의 요소들은 쉼표로 구분됩니다.  
집합은 수학에서의 집합 개념과 유사하게, 합집합, 교집합, 차집합 등의 연산을 제공합니다.

집합의 주요 특징:

중복 불허: 같은 값을 중복하여 가질 수 없습니다. 집합에 같은 값이 여러 번 추가되더라도, 집합 내에는 하나의 값만 유지됩니다.  
순서 없음(Unordered): 집합은 순서를 가지지 않습니다. 따라서, 인덱스를 통한 접근이 불가능합니다.

In [293]:
my_empty_set = set() # 빈 집합을 생성할 때 set을 붙여 주고 소괄호 사용, 반복 가능한 객체 들을 집합으로 만들때 사용

my_set = {1, 2, 3, 4, 5, 6, 7}

my_set2 = set([5, 6, 7, 8, 9, 10])

### 1. 인덱싱

집합 자료형은 딕셔너리와 같이 순서가 없기 때문에 인덱싱을 통해 요소값을 얻을 수 없다.

In [273]:
# my_set[0]

TypeError: 'set' object is not subscriptable

In [271]:
list_a = list(my_set)

list_a[:]

[1, 2, 3, 4, 5]

### 2. 집합 구하기

#### 1. 교집합

In [275]:
my_set & my_set2

{5, 6, 7}

#### 2. 합집합

In [276]:
my_set | my_set2

{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}

#### 3. 차집합

In [277]:
my_set - my_set2

{1, 2, 3, 4}

In [278]:
my_set2 - my_set

{8, 9, 10}

#### 4. 대칭 차집합

In [294]:
my_set ^ my_set2

{1, 2, 3, 4, 8, 9, 10}

| 연산 | 기호 | 설명 |
|------|------|------|
| 교집합 | `&` | 두 집합에 모두 있는 요소만 포함하는 새로운 집합을 반환 |
| 합집합 | `\|` | 두 집합의 모든 요소를 포함하는 새로운 집합을 반환, 중복은 제거됨 |
| 차집합 | `-` | 첫 번째 집합에는 있지만 두 번째 집합에는 없는 요소만 포함하는 새로운 집합을 반환 |
| 대칭 차집합 | `^` | 양쪽 집합 중 하나에만 있는 요소를 모두 포함하는 새로운 집합을 반환 |


### 3. 집합 함수

#### 1. add

In [285]:
my_set = {1, 2, 3, 4, 5, 6, 7}

my_set.add(8) # 값 1개 추가

my_set.add(1) # 중복 불허

print(my_set)

{1, 2, 3, 4, 5, 6, 7, 8}


#### 2. update

In [286]:
my_set = {1, 2, 3}

my_set.update([4, 5, 6]) # 여러 개 추가

print(my_set)

{1, 2, 3, 4, 5, 6}


#### 3. remove

In [42]:
my_set = {1, 2, 3}

my_set.remove(2) # 특정 값 제거

print(my_set)

{1, 3}


| 함수 | 설명 |
|------|------|
| `add(element)` | 집합에 요소를 추가합니다. s1.add(1) |
| `update(other)` | 집합에 다른 집합, 리스트, 튜플 등의 요소를 추가합니다. s1.update([4, 5, 6]) |
| `remove(element)` | 집합에서 요소를 제거합니다. 해당 요소가 집합에 없으면 KeyError를 발생시킵니다. s1.remove(2) |
| `discard(element)` | 집합에서 요소를 제거합니다. 해당 요소가 집합에 없어도 에러가 발생하지 않습니다. s1.discard(2) |
| `clear()` | 집합의 모든 요소를 제거합니다. s1.clear() |

## 9. 불(bool) 자료형

bool 자료형은 파이썬에서 논리적인 값을 나타내는 데 사용되며, True 또는 False 두 가지 값만을 가질 수 있습니다.  
bool은 불리언(Boolean) 자료형으로 불려지며, 조건문의 결과, 표현식의 평가, 객체의 상태 등을 표현하는 데 널리 사용됩니다.

bool 자료형의 주요 특징:

두 가지 값: True(참) 또는 False(거짓)만을 가질 수 있습니다.  
조건문과 논리 연산: bool 자료형은 조건문에서 조건의 참과 거짓을 판단하거나, 논리 연산(and, or, not)의 결과를 나타내는 데 사용됩니다.  
값의 진리성 평가: 파이썬에서는 bool 함수를 사용하여 다른 데이터 타입의 값을 True 또는 False로 변환할 수 있습니다. (일반적으로 0, None, 빈 컬렉션([], {}, (), '' 등)은 False로 평가되며, 이 외의 값은 True로 평가됩니다.)

| 값         | 불리언 평가 |
|------------|-----------|
| `True`     | True      |
| `False`    | False     |
| `0`        | False     |
| `1`        | True      |
| `-1`       | True      |
| `""` (빈 문자열) | False  |
| `" "` (공백) | True   |
| `[]` (빈 리스트) | False  |
| `[1, 2]` (요소가 있는 리스트) | True |
| `()` (빈 튜플) | False  |
| `(1,)` (요소가 있는 튜플) | True |
| `{}` (빈 딕셔너리) | False  |
| `{'key': 'value'}` (요소가 있는 딕셔너리) | True |
| `None`     | False     |
| `set()` (빈 집합) | False  |
| `{1, 2}` (요소가 있는 집합) | True |

bool()로 찍어볼 수 있음

In [49]:
bool(1) and bool(0)

False

bool(1) and bool(0)
| 불 연산자 | 설명                                                    |
|---------------|---------------------------------------------------------|
| `and`         | 두 조건이 모두 참(True)일 때만 참을 반환합니다.             |
| `or`          | 두 조건 중 하나라도 참(True)이면 참을 반환합니다.           |
| `not`         | 참(True)을 거짓(False)으로, 거짓을 참으로 반전시킵니다.      |

#### 중간 실습
a = True  
b = False  
and, or, not을 사용하여 다양한 값 찍어보기

In [50]:
a = [1, 2, 3, 4, 5]
while a:
    b = a.pop()
    print(b)

5
4
3
2
1


In [288]:
is_open = True

# 조건문에서 사용

if is_open:
  print('문이 열렸습니다')
else:
  print('문이 닫혔습니다')

문이 열렸습니다


In [289]:
foods = ['떡볶이', '초밥', '마라탕', '김치찌개', '라멘', '짜장면']

# 반복문에서 사용

while foods:
  print(foods.pop())

짜장면
라멘
김치찌개
마라탕
초밥
떡볶이


## 중간실습
1. [1, 2, 3, 4, 5], [4, 5, 6, 7, 8] 두 리스트에서 공통된 요소를 찾으세요
2. 1~10까지의 리스트의 요소들의 합계를 출력하세요 (sum 사용)
3. 본인의 이름, 생년월일, 주소지를 딕셔너리에 입력하고 모든 키와 값을 리스트로 변환 후 출력하세요
4. 1~10까지의 리스트를 3번 곱한 후 집합 자료형으로 변경해 보세요

In [51]:
#1
list1 = [1, 2, 3, 4, 5]
list2 = [4, 5, 6, 7, 8]
set1 = set(list1)
set2 = set(list2)
set3 = set1 & set2
print(set3)

{4, 5}


In [313]:
#2
numbers = list(range(1, 11))
print(f"합계: {sum(numbers)}")

합계: 55


In [48]:
#3
info = {
    "name": "김영욱",
    "birth": "1990-04-02",
    "address": "분당"
}
keys_list = list(info.keys())
values_list = list(info.values())
print(f"키 리스트: {keys_list}")
print(f"값 리스트: {values_list}")

키 리스트: ['name', 'birth', 'address']
값 리스트: ['김영욱', '1990-04-02', '분당']


In [53]:
#4
numbers = list(range(1, 11))
s1 = set(numbers * 3)
print(f"집합으로 변경: {s1}")

집합으로 변경: {1, 2, 3, 4, 5, 6, 7, 8, 9, 10}


## 자료형 정리

| 자료형 | 설명 | 예시 |
|--------|------|------|
| int | 정수형 데이터 | `123`, `-456` |
| float | 부동소수점 형태의 실수 데이터 | `3.14`, `-0.001` |
| str | 문자열 데이터 | `'hello'`, `"world"` |
| bool | 불리언 데이터, `True` 또는 `False` 값만을 가집니다. | `True`, `False` |
| list | 순서가 있는 항목들의 컬렉션, 변경 가능(mutable). | `[1, 2, 3]`, `['a', 'b', 'c']` |
| tuple | 순서가 있는 항목들의 컬렉션, 변경 불가능(immutable). | `(1, 2, 3)`, `('a', 'b', 'c')` |
| dict | 키와 값의 쌍으로 이루어진 컬렉션, 순서가 없음. | `{'key': 'value', 'id': 123}` |
| set | 중복되지 않는 항목들의 집합, 순서가 없음. | `{1, 2, 3}`, `{'a', 'b', 'c'}` |
